src/main/java/com/google/firebase/auth/AbstractFirebaseAuth.java

```diff
@@ -945,6 +936,98 @@ protected String execute() throws FirebaseAuthException {
     };
   }
 
+  /**
+   * Creates a new provider OIDC Auth config with the attributes contained in the specified {@link
+   * OidcProviderConfig.CreateRequest}.
+   *
+   * @param request A non-null {@link OidcProviderConfig.CreateRequest} instance.
+   * @return An {@link OidcProviderConfig} instance corresponding to the newly created provider
+   *     config.
+   * @throws NullPointerException if the provided request is null.
+   * @throws FirebaseAuthException if an error occurs while creating the provider config.
+   */
+  public OidcProviderConfig createOidcProviderConfig(
```

Comment:
This is different from the signature proposed in go/firebase-mt-java. That is fine, but we should think if this is the best we can do. In .NET we used a generics-based solution for this. Can we do something similar here? Something like the following:

```
<T extends ProviderConfig> T createProviderConfig(ProviderConfig.CreateRequest<T> request)
```

WDYT?

Reply:
It's unfortunately not as simple as this. The class signature of AuthProviderConfig is `public abstract static class CreateRequest<T extends CreateRequest<T>>`, not `public abstract static class CreateRequest<T extends AuthProviderConfig<T>>`. In order to make this work, I think we'd need to add another (complex) generic type to `AuthProviderConfig.CreateRequest`. As I mentioned in my PR description, the accumulated complexity outweighed the benefits in my opinion.

For more context about why `CreateRequest<T extends CreateRequest<T>>` is necessary (as opposed to the more intuitive `CreateRequest<T extends AuthProviderConfig>`), see https://stackoverflow.com/a/17165079.

Reply:
Wouldn't this work?

```
class CreateRequest<T extends CreateRequest<T>, U extends AuthProviderConfig> {
 
}
```

Reply:
Yes, I think that might work. That's the idea I was referring to when I mentioned adding another (complex) generic type to AuthProviderConfig.CreateRequest.

My concern is that this makes the external API too complex. Wouldn't this make the signature for `createProviderConfig` pretty wild?

Reply:
That's a good point. I guess the best we can do in that case is something like:

```
public <T extends ProviderConfig> T createProviderConfig(AbstractCreateRequest<?, T> request)
```

Users won't have to worry too much about this as they can simply do `createProviderConfig(new OidcProviderConfig.CreateRequest())` most of the time, but `createOidcProviderConfig(request)` is far simpler to wrap one's head around.

Ok, I think I'm sufficiently convinced. Let's go ahead with separate create methods for the 2 types.

Reply:
Great! Glad we are on the same page now.